### PR TITLE
feat(controls): Add react versions of time and volume scrubbers

### PR DESCRIPTION
**Pull request by jstoffan**

### CODE REVIEW

File: src/lib/viewers/controls/media/TimeControls.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
+@import '../slider/styles';
+
+$bp-TimeControls-size: 18px; // Divisible by 3px (default) and 6px (hovered) for better vertical centering
```
Comment (ChenCodes) [inline src/lib/viewers/controls/media/TimeControls.scss:3]: Is there some sort of grid styling that we can reuse here? For example, in box-ui-elements there is a variable called `$bdl-grid-unit` which is defaulted to 4px. Is the 3px above a constant that is being shared across multiple elements in the control bar?

File: src/lib/viewers/controls/media/TimeControls.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+export type Props = {
+    bufferedRange?: TimeRanges;
+    currentTime?: number;
+    durationTime?: number;
+    onTimeChange: (volume: number) => void;
+};
+
+export const round = (value: number): number => {
+    return +value.toFixed(4);
```
Comment (ChenCodes) [inline src/lib/viewers/controls/media/TimeControls.tsx:14]: Could you explain what the `+` symbol is for here?

File: src/lib/viewers/controls/slider/__tests__/SliderControl-test.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import SliderControl from '../SliderControl';
+
+describe('SliderControl', () => {
+    const getWrapper = (props = {}): ShallowWrapper =>
+        shallow(<SliderControl onChange={jest.fn()} value={0} {...props} />);
+
+    describe('event handlers', () => {
+        test('should parse and return the current value on change', () => {
+            const onChange = jest.fn();
+            const wrapper = getWrapper({ step: 0.1, onChange });
```
Comment (ChenCodes) [inline src/lib/viewers/controls/slider/__tests__/SliderControl-test.tsx:12]: Nit: abc

File: src/lib/viewers/controls/media/TimeControls.scss (hunk 1 above)
Comment (jstoffan) [reply src/lib/viewers/controls/media/TimeControls.scss:3]: No, we don't have such a system for the preview controls. We're trying to keep the visual styles as close as possible to what's live today. We can investigate refactoring the styles if and when we do a visual refresh in the future.

File: src/lib/viewers/controls/media/TimeControls.tsx (hunk 2 above)
Comment (jstoffan) [reply src/lib/viewers/controls/media/TimeControls.tsx:14]: It's a shortcut to convert a string to a number, as the former is what `toFixed` returns.

File: src/lib/viewers/controls/media/TimeControls.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    const handleChange = (newValue: number): void => {
+        onTimeChange(round(durationTime * (newValue / 100)));
+    };
+
+    return (
+        <SliderControl
+            className="bp-TimeControls"
+            onChange={handleChange}
+            step={0.1}
```
Comment (ChenCodes) [inline src/lib/viewers/controls/media/TimeControls.tsx:39]: It looks like we have this step property that is being passed in to SliderControl, but later in the SliderControl.tsx file, we're hardcoding step={1}. Does this prop still need to be passed down?

Comment (jstoffan) [reply src/lib/viewers/controls/media/TimeControls.tsx:39]: Yes, as it overrides the default value of `1` in SliderControl.

**Review by ChenCodes — APPROVED** [review]
LGTM

File: src/lib/viewers/controls/slider/__tests__/SliderControl-test.tsx (hunk 3 above)
Comment (jstoffan) [reply src/lib/viewers/controls/slider/__tests__/SliderControl-test.tsx:12]: Will fix in a follow-up.

**Pull request merged**